src/sknnr/_base.py

```diff
@@ -60,3 +80,15 @@ def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
         return super().kneighbors(
             X=X_transformed, n_neighbors=n_neighbors, return_distance=return_distance
         )
+
+
+@contextmanager
+def set_temp_output(estimator: BaseEstimator, temp_mode: Literal["default", "pandas"]):
```

Comment:
Actually, I think moving it into `transformers._base` and renaming the argument would be helpful for me for understanding what estimators it applies to.  I can't say that I fully understand this 100% yet (as evidenced by my comments), but I'll reread and hopefully get there!

Reply:
Will do!

It took me a long time working through this PR to wrap my head around how the data moves through the estimators and transformers and where feature names get accessed and data types get changed, and I still wouldn't say I understand every edge case. I'd be pretty impressed if you could grok it in one morning! I think the truth is that even though we haven't written that much code, we're inheriting a lot of complexity from sklearn, which makes this a pretty challenging project.

Let me know if you run into anything particularly confusing and (assuming I understand it) I can try to explain or add some more comments to the code.

Reply:
Thanks for going easy on me :wink:.  For me, it's pretty useful to see when and where tests fail (i.e. in #21) because that leads down into the guts of `sklearn` and provides some good jumping off points.  It doesn't necessarily give me inspiration on how to *fix* things, but I generally have a better sense of what is happening.  Keep going, and I'll catch up!